### PR TITLE
Set Minitest.seed when needed

### LINKED ISSUE #99: `vscode:minitest:list` -> TypeError: no implicit conversion of nil into Integer
### Your environment

- `vscode-ruby-test-adapter` version: v0.9.0
- Ruby version: 2.7.6p219
- VS Code version: 1.69.1
- Operating System: macOS Monterey 12.4
- RSpec or Minitest version: Minitest 5.16.2

### Expected behaviour

Typically a few seconds pass and my tests load. When they don't it is usually because I have a `rake` error...e.g. my postgres container isn't running the app's database. At this point I'd start the database and perform a VS Code `Developer: Reload Window`. 

### Actual behaviour

In my case, I have two separate Rails applications who are suddenly failing to list tests even when everything seems to be working. I do have some _rake_ noise where some Ruby/Rails trace/deprecation messages are appearing but they are in no way producing a non-zero error code preventing the `vscode:minitest:list` command from succeeding.

Here are the log messages from the extension; note the **no implicit conversion of nil into Integer** ... I'm seeing this in multiple projects.

```
[2022-07-12 20:24:25.727] [INFO] Getting a list of Minitest tests in suite with the following command: DISABLE_SPRING=1 PARALLEL_WORKERS=1 ./bin/rake -R $EXT_DIR vscode:minitest:list
[2022-07-12 20:24:25.727] [INFO] Initialization finished
[2022-07-12 20:24:30.287] [ERROR] Error while finding Minitest test suite: Command failed: DISABLE_SPRING=1 PARALLEL_WORKERS=1 ./bin/rake -R $EXT_DIR vscode:minitest:list
... (I have some deprecation messages from rails here BUT nothing generating an error code) ... 
rake aborted!
TypeError: no implicit conversion of nil into Integer
/Users/.../.vscode/extensions/connorshea.vscode-ruby-test-adapter-0.9.0/ruby/vscode/minitest/tests.rb:33:in `block in build_list'
/Users/.../.vscode/extensions/connorshea.vscode-ruby-test-adapter-0.9.0/ruby/vscode/minitest/tests.rb:32:in `map'
/Users/.../.vscode/extensions/connorshea.vscode-ruby-test-adapter-0.9.0/ruby/vscode/minitest/tests.rb:32:in `build_list'
/Users/.../.vscode/extensions/connorshea.vscode-ruby-test-adapter-0.9.0/ruby/vscode/minitest/tests.rb:9:in `all'
/Users/.../.vscode/extensions/connorshea.vscode-ruby-test-adapter-0.9.0/ruby/vscode/minitest.rb:25:in `list'
/Users/.../.vscode/extensions/connorshea.vscode-ruby-test-adapter-0.9.0/ruby/vscode.rake:8:in `block (3 levels) in <main>'
Tasks: TOP => vscode:minitest:list
(See full trace by running task with --trace)
```

### PR DESCRIPTION
**Pull request by blowmage**
Ensure a seed is present when running Minitest tests.

[fixes #99]

### CODE REVIEW

File: ruby/vscode/minitest/tests.rb [hunk 1]
```diff
@@ -28,6 +28,10 @@ def load_files
       end
 
       def build_list
+        if Minitest.respond_to?(:seed) && Minitest.seed.nil?
+          Minitest.seed = (ENV['SEED'] || srand).to_i % 0xFFFF
+        end
```
Comment (nadnoslen) [inline ruby/vscode/minitest/tests.rb:33]: Terrific!

**Comment by pnomolos** [comment]
This needs to be updated to the following (I shortened the syntax, but the important bit is the root-level module reference):

```ruby
::Minitest.seed ||= (ENV["SEED"] || srand).to_i % 0xFFFF if ::Minitest.respond_to?(:seed)
```

Otherwise it just doesn't work, because `VSCode::Minitest` get resolved before `Minitest` and it doesn't have a `seed` method, so the conditional fails.

**Comment by blowmage** [comment]
Updated.

**Comment by SergeyBurtsev** [comment]
Just a thought: maybe we can define the sorted order for tests instead of random order when building the list before calling #runnable_methods and thereby remove the dependency on seed completely? And then we don't need to sort the tests by their line numbers afterwards. What do you think?

```
def build_list
  tests = []
  ::Minitest::Runnable.runnables.map do |runnable|
    runnable.define_singleton_method(:test_order) { :sorted }
    file_tests = runnable.runnable_methods.map do |test_name|
      ...
    end
    # file_tests.sort_by! { |t| t[:line_number] }
    ...
  end
end
```

**Comment by pnomolos** [comment]
The whole point of ordering a test suite randomly is to discover issues caused by hidden dependencies due to tests being run in a certain order, so my vote for that would be no.

**Comment by SergeyBurtsev** [comment]
Good point @pnomolos . Thought, maybe build_list method used for building the list of tests only and running those tests uses different flow, but I double checked and see, that build_list is used for running tests as well, so tweaking test_order for runnables is not good.

**Comment by blowmage** [comment]
Ping. Anything else needed to get this change accepted?

**Comment by connorshea** [comment]
Nope, change seems reasonable to me. I'll run CI and assuming that passes, I'll merge and cut a new release.

**Comment by connorshea** [comment]
Spoken too soon :( It looks like the tests are failing on main as well, though, so let me see if I can figure that out.

**Comment by connorshea** [comment]
...so I guess this changed solved a secondary problem in CI, so now CI passes. Neat, didn't expect that. Thank you!

**Comment by blowmage** [comment]
Thanks!

**Comment by finnove** [comment]
Great! Thank you! This seem to have solved my problem in https://github.com/minitest/minitest/issues/917
(running minitest 5.16.3 this time, but I think this seed fix was the crucial one 👍)

**Pull request merged**